<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowTitle">
   <string>LorettLink — Приёмник</string>
  </property>
  <property name="minimumSize">
   <size><width>900</width><height>640</height></size>
  </property>

  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="rootLayout">
    <property name="spacing"><number>6</number></property>
    <property name="leftMargin"><number>6</number></property>
    <property name="topMargin"><number>6</number></property>
    <property name="rightMargin"><number>6</number></property>
    <property name="bottomMargin"><number>6</number></property>

    <!-- Connection: COM + TCP -->
    <item>
     <widget class="QFrame" name="frameConnection">
      <property name="frameShape"><enum>QFrame::StyledPanel</enum></property>
      <layout class="QHBoxLayout" name="connLayout">
       <property name="leftMargin"><number>12</number></property>
       <property name="topMargin"><number>8</number></property>
       <property name="rightMargin"><number>12</number></property>
       <property name="bottomMargin"><number>8</number></property>
       <property name="spacing"><number>12</number></property>

       <item>
        <widget class="QLabel"><property name="text"><string>COM:</string></property></widget>
       </item>
       <item>
        <widget class="QComboBox" name="cb_port">
         <property name="minimumSize"><size><width>100</width><height>0</height></size></property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="cb_baud">
         <property name="currentText"><string>115200</string></property>
         <additem><property name="text"><string>9600</string></property></additem>
         <additem><property name="text"><string>115200</string></property></additem>
         <additem><property name="text"><string>230400</string></property></additem>
         <additem><property name="text"><string>460800</string></property></additem>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btn_refresh">
         <property name="text"><string>Обновить</string></property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btn_connect">
         <property name="text"><string>Подключить</string></property>
        </widget>
       </item>
       <item>
        <widget class="QLabel"><property name="text"><string>TCP:</string></property></widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sb_tcp_port">
         <property name="minimum"><number>1024</number></property>
         <property name="maximum"><number>65535</number></property>
         <property name="value"><number>12000</number></property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btn_tcp">
         <property name="text"><string>Слушать</string></property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btn_save">
         <property name="text"><string>Сохранить</string></property>
         <property name="enabled"><bool>false</bool></property>
        </widget>
       </item>
       <item>
        <spacer><property name="orientation"><enum>Qt::Horizontal</enum></property>
         <property name="sizeHint" stdset="0"><size><width>20</width><height>20</height></size></property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>

    <!-- Telem: RSSI / SNR / TX Power -->
    <item>
     <widget class="QFrame" name="frameTelem">
      <property name="frameShape"><enum>QFrame::StyledPanel</enum></property>
      <layout class="QHBoxLayout" name="telemLayout">
       <property name="leftMargin"><number>12</number></property>
       <property name="topMargin"><number>4</number></property>
       <property name="rightMargin"><number>12</number></property>
       <property name="bottomMargin"><number>4</number></property>
       <property name="spacing"><number>12</number></property>
       <item>
        <widget class="QLabel" name="lbl_rssi">
         <property name="text"><string>RSSI: — дБм</string></property>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="bar_rssi">
         <property name="minimum"><number>-140</number></property>
         <property name="maximum"><number>-50</number></property>
         <property name="value"><number>-100</number></property>
         <property name="textVisible"><bool>false</bool></property>
         <property name="maximumSize"><size><width>120</width><height>20</height></size></property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="lbl_snr">
         <property name="text"><string>SNR: — дБ</string></property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="lbl_txpower">
         <property name="text"><string>TX: — дБм</string></property>
        </widget>
       </item>
       <item>
        <spacer><property name="orientation"><enum>Qt::Horizontal</enum></property>
         <property name="sizeHint" stdset="0"><size><width>20</width><height>20</height></size></property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>

    <!-- Splitter: matrix | image -->
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation"><enum>Qt::Horizontal</enum></property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch><verstretch>1</verstretch>
       </sizepolicy>
      </property>

      <widget class="QWidget" name="leftPanel">
       <layout class="QVBoxLayout" name="leftLayout">
        <property name="leftMargin"><number>0</number></property>
        <property name="topMargin"><number>0</number></property>
        <property name="rightMargin"><number>0</number></property>
        <property name="bottomMargin"><number>0</number></property>
        <item>
         <widget class="QLabel">
          <property name="text"><string>Матрица блоков</string></property>
          <property name="alignment"><set>Qt::AlignCenter</set></property>
          <property name="font"><font><family>Segoe UI</family><pointsize>10</pointsize><bold>true</bold></font></property>
         </widget>
        </item>
        <item>
         <widget class="ChunkMatrixWidget" name="matrix">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>1</horstretch><verstretch>1</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize"><size><width>200</width><height>140</height></size></property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lbl_chunks">
          <property name="text"><string>Ожидание FEC...</string></property>
          <property name="alignment"><set>Qt::AlignCenter</set></property>
         </widget>
        </item>
       </layout>
      </widget>

      <widget class="QWidget" name="rightPanel">
       <layout class="QVBoxLayout" name="rightLayout">
        <property name="leftMargin"><number>0</number></property>
        <property name="topMargin"><number>0</number></property>
        <property name="rightMargin"><number>0</number></property>
        <property name="bottomMargin"><number>0</number></property>
        <item>
         <widget class="QLabel">
          <property name="text"><string>Изображение</string></property>
          <property name="alignment"><set>Qt::AlignCenter</set></property>
          <property name="font"><font><family>Segoe UI</family><pointsize>10</pointsize><bold>true</bold></font></property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="img_label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>1</horstretch><verstretch>1</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize"><size><width>200</width><height>200</height></size></property>
          <property name="alignment"><set>Qt::AlignCenter</set></property>
          <property name="styleSheet"><string notr="true">background:#1a1a1a; border:1px solid #444; border-radius:4px;</string></property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>

    <!-- Progress -->
    <item>
     <widget class="QProgressBar" name="progress">
      <property name="minimum"><number>0</number></property>
      <property name="maximum"><number>1</number></property>
      <property name="value"><number>0</number></property>
      <property name="textVisible"><bool>true</bool></property>
      <property name="format"><string>  %p%  —  %v / %m блоков</string></property>
     </widget>
    </item>

    <!-- Log -->
    <item>
     <widget class="QTextEdit" name="log">
      <property name="readOnly"><bool>true</bool></property>
      <property name="maximumSize"><size><width>16777215</width><height>140</height></size></property>
      <property name="font"><font><family>Consolas</family><pointsize>9</pointsize></font></property>
     </widget>
    </item>

   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>

 <customwidgets>
  <customwidget>
   <class>ChunkMatrixWidget</class>
   <extends>QWidget</extends>
   <header>widgets</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
